.github/workflows/codeql-analysis.yml

```diff
@@ -51,13 +51,12 @@ jobs:
         run: |
           poetry install --no-root --all-extras --with dev --without docs,test
 
-      - name: Run flake8
-        run: poetry run flake8 data_rentgen/
+      - name: Run ruff
+        run: poetry run ruff check --output-format=github data_rentgen/
```

Comment:
This was run by Github Actions and not pre-commit.ci because flake8 requried a lot of plugins. ruff doesn't have any plugins, so this can be moved to pre-commit.ci